intellij-plugin-structure/structure-base/src/main/kotlin/com/jetbrains/plugin/structure/base/problems/InvalidDescriptorErrors.kt

```diff
@@ -136,3 +136,41 @@ class VendorCannotBeEmpty(
   override val level
     get() = Level.ERROR
 }
+
+class InvalidSemverFormat(
+  descriptorPath: String,
+  versionName: String,
+  version: String
+) : InvalidDescriptorProblem(
+  descriptorPath = descriptorPath,
+  detailedMessage = "The `$versionName` version should be formatted as semver [$version]."
+) {
+  override val level
+    get() = Level.ERROR
+}
+
+class InvalidVersionRange(
+  descriptorPath: String,
+  since: String,
+  until: String
+) : InvalidDescriptorProblem(
+  descriptorPath = descriptorPath,
+  detailedMessage = "The compatibility range [$since, $until] is empty."
+) {
+  override val level
+    get() = Level.ERROR
+}
+
+class SemverComponentLimitExceeded(
+  descriptorPath: String,
+  componentName: String,
+  versionName: String,
+  version: String,
+  limit: Int
+) : InvalidDescriptorProblem(
+  descriptorPath = descriptorPath,
+  detailedMessage = "The `$componentName` component of the `$versionName` semver version is too big [$version]. Max value is $limit."
```

Comment:
In the message, it's better to use `SemVer` (as per original [document](https://semver.org/)) or full _Semantic Versioning_.